src/site/content/en/blog/extending-workbox/index.md

```diff
@@ -0,0 +1,379 @@
+---
+title: Extending Workbox
+subhead: >
+  Write your own reusable, shareable strategies and plugins.
+description: >
+  Write your own reusable, shareable strategies and plugins.
+date: 2020-12-10
+thumbnail: thumbnail.png
+hero: hero.png
+alt: The Workbox logo.
+draft: true
+tags:
+  - blog
+  - service-workers
+---
+
+In this article, we're going to take a quick tour of some ways of extending
+[Workbox](https://developers.google.com/web/tools/workbox). By the end, you'll
+be writing your own strategies and plugins, and hopefully sharing them with the
+world.
+
+<!--
+If you're more of a visual person, you can watch a recording of a Chrome
+Dev Summit talk covering the same material:
+
+{% YouTube 'your-video-id' %}
+-->
+
+## What's Workbox?
+
+At its core, Workbox is a set of libraries to help with common service worker
+caching scenarios. And when we've written about Workbox in the past, the
+emphasis has been on "common" scenarios. For most developers, the
+[caching strategies](https://developers.google.com/web/tools/workbox/modules/workbox-strategies)
+that Workbox already provides will handle your caching needs.
+
+The built-in strategies include
+[stale-while-revalidate](https://developers.google.com/web/tools/workbox/modules/workbox-strategies#stale-while-revalidate),
+where a cached response is used to respond to a request immediately, while the
+cache is also updated so that it's fresh the next time around. They also include
+[network-first](https://developers.google.com/web/tools/workbox/modules/workbox-strategies#network_first_network_falling_back_to_cache),
+falling back to the cache when the network is unavailable, and a few more.
+
+## Custom strategies
+
+But what if you wanted to go beyond those common caching scenarios? Let's cover
+writing your own custom caching strategies.
+[Workbox v6](https://github.com/GoogleChrome/workbox/releases/tag/v6.0.0) offers
+a new `Strategy` base class that sits in front of lower-level APIs, like
+[Fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) and
+[Cache Storage](https://developer.mozilla.org/en-US/docs/Web/API/CacheStorage).
+You can extend the `Strategy` base class, and then implement your own logic in
+the `_handle()` method.
+
+### DedupeNetworkFirst
```

Comment:
```suggestion
### Handle simultaneous, duplicate requests with DedupeNetworkFirst
```
Didn't catch this during the early review: Let's update these sections to follow the "X with Y" pattern, where X is the general task that's solved and Y is the name of the feature?